gallery/__init__.py

```diff
@@ -791,7 +797,17 @@ def display_dir_thumbnail(dir_id: int, auth_dict: Optional[Dict[str, Any]] = Non
         thumbnail_uuid = thumbnail_uuid.split('.')[0]
 
     link = storage_interface.get_link("thumbnails/{}".format(thumbnail_uuid))
-    return redirect(link)
+    if "LOCAL_STORAGE_PATH" in app.config:
+        link = "http://" + app.config["SERVER_NAME"] + link
+    req = requests.get(link)
+    while req.status_code != requests.codes.ok:
+        dir_model.thumbnail_uuid = refresh_directory_thumbnail(dir_model)
```

Comment:
Will this default to the "no thumbnail" photo if it fails to generate one? Want to be sure this can't loop infinitely.

Reply:
Yes